arviz/data/inference_data.py

```diff
@@ -324,6 +325,59 @@ def sel(
         else:
             return out
 
+    def add_groups(self, group_dict=None, coords=None, dims=None, **kwargs):
+        """Add new groups to InferenceData object.
+
+        Parameters
+        ----------
+        group_dict: dict[str] -> dict, xr.Dataset
+            Groups to be added
+        coords : dict[str] -> ndarray
+            Coordinates for the dataset
+        dims : dict[str] -> list[str]
+            Dimensions of each variable. The keys are variable names, values are lists of
+            coordinates.
+        **kwargs: mapping
+            The keyword arguments form of group_dict. One of group_dict or kwargs must be provided.
+
+        """
+        if group_dict is None:
+            group_dict = {}
+
+        group_dict.update(kwargs)
```

Comment:
I would try to mimic xarray, may be even use [this same function](https://github.com/pydata/xarray/blob/1be777fe725a85b8cc0f65a2bc41f4bc2ba18043/xarray/core/utils.py#L256). This implementation behaves differently from xarray in the sense that kwargs overwrite group_dict whereas in xarray both of them being present raises an error

Reply:
see for example [isel](https://github.com/pydata/xarray/blob/master/xarray/core/dataarray.py#L1042) as an example of how dict/kwargs are implemented